wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+        auth_api.create_namespaced_role_binding(namespace=namespace, body=builder_role_binding)
+
+        # adding deployer role binding
+        deployer_role = self.kclient.V1ObjectReference(name='system:deployer')
+        deployer_sa = self.kclient.V1ObjectReference(name='deployer',
+                                                     kind='ServiceAccount',
+                                                     namespace=namespace)
+        deployer_role_binding_name = self.kclient.V1ObjectMeta(name='deployer-binding')
+        deployer_role_binding = self.ociclient.V1RoleBinding(role_ref=deployer_role,
+                                                             subjects=[deployer_sa],
+                                                             metadata=deployer_role_binding_name)
+        auth_api.create_namespaced_role_binding(namespace=namespace, body=deployer_role_binding)
+
+        # adding admin role binding
+        admin_role = self.kclient.V1ObjectReference(name='admin')
+        admin_user = self.kclient.V1ObjectReference(name='admin',
+                                                    kind='User',
+                                                    namespace=namespace)
+        admin_role_binding_name = self.kclient.V1ObjectMeta(name='admin-binding')
+        admin_role_binding = self.ociclient.V1RoleBinding(role_ref=admin_role,
+                                                          subjects=[admin_user],
+                                                          metadata=admin_role_binding_name)
+        auth_api.create_namespaced_role_binding(namespace=namespace, body=admin_role_binding)
+
+        # adding image-puller role binding
+        puller_role = self.kclient.V1ObjectReference(name='system:image-puller')
+        group_name = 'system:serviceaccounts:{proj}'.format(proj=namespace)
+        puller_group = self.kclient.V1ObjectReference(name=group_name,
+                                                      kind='SystemGroup',
+                                                      namespace=namespace)
+        role_binding_name = self.kclient.V1ObjectMeta(name='image-puller-binding')
+        puller_role_binding = self.ociclient.V1RoleBinding(role_ref=puller_role,
+                                                           subjects=[puller_group],
+                                                           metadata=role_binding_name)
+        auth_api.create_namespaced_role_binding(namespace=namespace, body=puller_role_binding)
+
+    def delete_project(self, name, wait=120):
+        """Removes project(namespace) and all entities in it.
+
+        Args:
+            name: project name
+            wait: within this time project should disappear
+        Return: None
+        """
+        self.logger.info("removing project %s", name)
+        if self.does_project_exist(name=name):
+            self.o_api.delete_project(name=name)
+            if not self.does_project_exist(name=name, wait=wait):
```

Comment:
__REQUIRED__
`does_project_exist` wait until the project exists, hence it'll either return immediately with True (because it didn't removed the project yet) and you wont know if the project removed  or  it'll wait 120 seconds for the project to be exists and it'll return False (and then it'll raise TimedOutError). I'd add `not_case` flag to `_does_exist` function to test non-existence as well.

Reply:
fixed